### PR TITLE
Ensure xStart, xEnd in bounds

### PR DESCRIPTION
**Pull request by DrJosh9000**
And xStart >= xEnd, etc

### CODE REVIEW

File: screen.go [hunk 1]
```diff
@@ -27,17 +27,31 @@ const screenEndOfLine = -1
 const screenStartOfLine = 0
 
 // Clear part (or all) of a line on the screen
-func (s *screen) clear(y int, xStart int, xEnd int) {
+func (s *screen) clear(y, xStart, xEnd int) {
 	if len(s.screen) <= y {
 		return
 	}
 
 	line := &s.screen[y]
```
Comment (triarius) [inline screen.go:35]: Does this blow up for `y < 0`?

**Review by triarius — COMMENTED** [review]
How about handling all the edge cases as early as possible? The ones I can think of are:

- `y < 0`
- `y >= len(s.screen)`
- `xStart < 0`
- `xStart >= len(line.nodes)`
- `xEnd < xStart`
- `xEnd >= len(line.nodes)` <- this one does not need explicit handling

File: screen.go (hunk 1 above)
Comment (DrJosh9000) [reply screen.go:35]: Almost certainly!

**Comment by DrJosh9000** [comment]
> How about handling all the edge cases as early as possible?

xEnd is complicated because screenEndOfLine is -1. Otherwise I've tried to do that.

**Comment by DrJosh9000** [comment]
> > How about handling all the edge cases as early as possible?
> 
> xEnd is complicated because screenEndOfLine is -1. 

Thinking further on this, the answer to that is to change it to something positive.

**Review by triarius — APPROVED** [review]
LGTM 🎉

**Pull request merged**